pkg/apigw/helpers.go

```diff
@@ -21,3 +25,35 @@ func helperDefaultResponse(opt *options.ApigwOpt) http.HandlerFunc {
 		}
 	}
 }
+
+func helperMethodNotAllowed(opt *options.ApigwOpt, pr *profiler.Profiler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		addToProfiler(opt, pr, r, http.StatusMethodNotAllowed)
+
+		if opt.LogEnabled {
+			// Say something friendly when logging is enabled
+			http.Error(w, devHelperResponseBody, http.StatusTeapot)
+		} else {
+			// Default 405 response
+			http.Error(w, "", http.StatusMethodNotAllowed)
+		}
+	}
+}
+
+func addToProfiler(opt *options.ApigwOpt, pr *profiler.Profiler, r *http.Request, status int) {
+	if !(opt.ProfilerEnabled && opt.ProfilerGlobal) {
+		return
+	}
+
+	// add to profiler
+	ar, err := h.NewRequest(r)
+
+	if err != nil {
+		panic(err)
```

Comment:
Can this situation be caused by a non-developer (so anyone w/o access to server source)?
If yes, then this should be converted to a log (zap).

Reply:
Did not want to wrap with logger and the rationale was that something really wrong should go when creating a wrapper request from `http.Request`, but there could be some issues with ie headers or body, so I'll revert that panic, :+1: